app/src/main/java/com/mealsmadeeasy/ui/home/profile/ProfileFragment.kt

```diff
@@ -0,0 +1,203 @@
+                              savedInstanceState: Bundle?): View? {
+        val root = inflater.inflate(R.layout.view_user_profile, container, false)
+
+        sexSpinner = root.findViewById(R.id.sex_spinner)
+        bdayPicker = root.findViewById(R.id.birthday_edit_text)
+        feetText = root.findViewById(R.id.height_feet_edit_text)
+        inchesText = root.findViewById(R.id.height_inches_edit_text)
+        weightText = root.findViewById(R.id.weight_field)
+        submitButton = root.findViewById(R.id.save_button)
+
+        // Populate sex spinner values
+        var sexValues: List<String> = emptyList()
+        for (item in Gender.valuesAsStringRes()) {
+            sexValues += getString(item)
+        }
+        sexValues += getString(R.string.profile_spinner_prompt)
+        val sexAdapter = HintAdapter(this.activity, android.R.layout.simple_list_item_1, sexValues)
+        sexAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        sexSpinner.adapter = sexAdapter
+        sexSpinner.setSelection(sexAdapter.count)
+
+        // Set up birthday date picker
+        val dateText = root.findViewById<EditText>(R.id.birthday_edit_text)
+        val cal = Calendar.getInstance()
+        setupDatePicker(dateText, cal)
+
+        // Set up text input view hints
+        val feetContainer = root.findViewById<TextInputLayout>(R.id.height_feet_container)
+        val inchesContainer = root.findViewById<TextInputLayout>(R.id.height_inches_container)
+        val weightContainer = root.findViewById<TextInputLayout>(R.id.weight_container)
+
+        feetText.onFocusChangeListener = TextInputViewHintListener(feetContainer,
+                getString(R.string.profile_height_prompt), getString(R.string.profile_height_feet_units))
+        inchesText.onFocusChangeListener = TextInputViewHintListener(inchesContainer,
+                getString(R.string.profile_height_inches_prompt), getString(R.string.profile_height_inches_units))
+        weightText.onFocusChangeListener = TextInputViewHintListener(weightContainer,
+                getString(R.string.profile_weight_prompt), getString(R.string.profile_weight_lbs_units))
+
+        // Enable button iff all fields are empty
+        val editTextList = listOf(bdayPicker, feetText, inchesText, weightText)
+        sexSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+                submitButton.isEnabled = areAllFieldsEnabled()
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>) {
+                ;
+            }
+        }
+
+        val watcher = ButtonTextWatcher()
+        for (editText in editTextList) {
+            editText.addTextChangedListener(watcher)
+        }
+
+        // Set button on click behavior
+        submitButton.setOnClickListener{
```

Comment:
This lambda might make sense to be a private function here (maybe `onSubmitClicked()`)